tasks/tomcat_wars.yml

```diff
@@ -0,0 +1,29 @@
+---
+- name: "Ensure webapp dir exists (url)"
+  shell: "mkdir -p {{ item.directory | default(tomcat_directory) }}/{{ tomcat_archive[item.version | default (tomcat_version)]['dir'] }}/webapps"
+  when:
+    - "tomcat_wars.url is defined"
+  with_items: "{{ tomcat_wars.url }}"
+- name: "loop over tomcat_wars"
+  get_url:
+    dest: "{{ item.directory | default(tomcat_directory) }}/{{ tomcat_archive[item.version | default (tomcat_version)]['dir'] }}/webapps/"
+    url: "{{ item }}"
+    validate_certs: false
+  when:
+    - "tomcat_wars.url is defined"
+  with_items: "{{ tomcat_wars.url }}"
+- name: "Ensure webapp dir exists (s3)"
+  shell: "mkdir -p {{ item.directory | default(tomcat_directory) }}/{{ tomcat_archive[item.version | default (tomcat_version)]['dir'] }}/webapps"
+  when:
+    - "tomcat_wars.s3 is defined"
+  with_items: "{{ tomcat_wars.s3 }}"
+- name: "loop over tomcat_wars in s3"
+  aws_s3:
```

Comment:
By the way: If the bucket is publicly available, the `url:` could be used to deploy the war right?

Reply:
Yes. But this lets you deploy out of private s3 buckets, and not make your war files readable to the world. Pretty important when your war contains sensitive or proprietary data.